File: source_code/R/get_data.R
library(readr)
library(dtplyr) 
library(dplyr)

ROOT <- rprojroot::find_rstudio_root_file()
DATA_PATH <- paste0(ROOT, "/data/")

get_movielens_small_ratings <- function(){
  read_csv(paste0(DATA_PATH, "movielens/small/ratings.csv"))
}

get_movielens_small_movies <- function(){
  read_csv(paste0(DATA_PATH, "movielens/small/movies.csv"))
}

get_movielens_small_tags <- function(){
  read_csv(paste0(DATA_PATH, "movielens/small/tags.csv"))
}

get_movielens_small <- function(){
  df_ratings <- get_movielens_small_ratings() %>% 
    lazy_dt()
    select(userId, movieId, rating)
  
  df_movies <- get_movielens_small_movies()

  df_ratings %>%
    left_join(df_movies, by = "movieId") %>% 
    select(userId, movieId, title, rating)
}

get_movielens_medium_ratings <- function(){
  read_csv(paste0(DATA_PATH, "movielens/medium/ratings.csv"))
}

get_movielens_medium_movies <- function(){
  read_csv(paste0(DATA_PATH, "movielens/medium/movies.csv"))
}

get_movielens_medium_tags <- function(){
  read_csv(paste0(DATA_PATH, "movielens/medium/tags.csv"))
}

get_movielens_medium <- function(){
  df_ratings <- get_movielens_medium_ratings() %>% 
    select(userId, movieId, rating)
  
  df_movies <- get_movielens_medium_movies()
  
  df_ratings %>%
    left_join(df_movies, by = "movieId") %>% 
    select(userId, movieId, title, rating)
}

get_movielens_large_ratings <- function(){
  read_csv(paste0(DATA_PATH, "movielens/large/ratings.csv"))
}

get_movielens_large_movies <- function(){
  read_csv(paste0(DATA_PATH, "movielens/large/movies.csv"))
}

get_movielens_large_tags <- function(){
  read_csv(paste0(DATA_PATH, "movielens/large/tags.csv"))
}

get_movielens_large <- function(){
  df_ratings <- get_movielens_large_ratings() %>% 
    select(userId, movieId, rating)
  
  df_movies <- get_movielens_large_movies()
  
  df_ratings %>%
    left_join(df_movies, by = "movieId") %>% 
    select(userId, movieId, title, rating)
}

File: source_code/Python/get_data.py
import numpy as np
import pandas as pd

MOVIELENS_PATH = "../data/movielens/"

# MovieLens -------------------

## Small -----------------------
def get_movielens_small_ratings():
    return pd.read_csv(MOVIELENS_PATH + "small/ratings.csv")

def get_movielens_small_movies():
    return pd.read_csv(MOVIELENS_PATH + "small/movies.csv")

def get_movielens_small():
    df_ratings = get_movielens_small_ratings()[["userId", "movieId", "rating"]]
    df_movies = get_movielens_small_movies()[["movieId", "title"]]
    
    df_ratings = df_ratings.merge(
        df_movies,
        on = "movieId"
    )
    
    return df_ratings

## Medium ------------------------
def get_movielens_medium_ratings():
    return pd.read_csv(MOVIELENS_PATH + "medium/ratings.csv")

def get_movielens_small_movies():
    return pd.read_csv(MOVIELENS_PATH + "medium/movies.csv")

def get_movielens_medium():
    df_ratings = get_movielens_medium_ratings()[["userId", "movieId", "rating"]]
    df_movies = get_movielens_medium_movies()[["movieId", "title"]]
    
    df_ratings = df_ratings.merge(
        df_movies,
        on = "movieId"
    )
    
    return df_ratings

## Large -------------------------
def get_movielens_large_ratings():
    return pd.read_csv(MOVIELENS_PATH + "large/ratings.csv")

def get_movielens_large_movies():
    return pd.read_csv(MOVIELENS_PATH + "large/movies.csv")

def get_movielens_large():
    df_ratings = get_movielens_large_ratings()[["userId", "movieId", "rating"]]
    df_movies = get_movielens_large_movies()[["movieId", "title"]]
    
    df_ratings = df_ratings.merge(
        df_movies,
        on = "movieId"
    )
    
    return df_ratings
    
   